babelsberg/constraintinterpreter.js

```diff
@@ -381,6 +336,110 @@ Object.subclass('Babelsberg', {
         return result;
     },
 
+    /**
+     * Create a Constraint for opts and func for each of the specified solvers.
+     * Return an array of the created Constraints.
+     */
+    createEquivalentConstraints: function(solvers, opts, func, errors) {
+        var constraints = [];
+        solvers.each(function(solver) {
+            try {
+                var optsForSolver = Object.clone(opts);
+                var constraint = solver.always(optsForSolver, func);
+                constraint.opts = optsForSolver;
+                constraint.originalOpts = opts;
+                constraints.push(constraint);
+            } catch (e) {
+                errors.push(e);
+                return;
+            }
+        });
+        return constraints;
+    },
+
+    /**
+     * Choose one of the specified constraints which performs best according to the
+     * requirements laid out in opts.
+     */
+    chooseConstraint: function(constraints, opts, errors) {
+        if (constraints.length === 1)
+            return constraints[0];
+        var constraint = null;
+        var previouslyEnabledConstraints = [];
+        // make sure all constraints are disabled before the comparison
+        constraints.each(function(each) {
+            if (each._enabled)
+                previouslyEnabledConstraints.push(each);
+            each.disable();
+        });
+        for (var i = 0; i < constraints.length; i++) {
+            try {
+                Constraint.current = constraints[i];
+                constraints[i].enable(true);
+                constraints[i].disable();
+            } catch (e) {
+                errors.push(e);
+                constraints[i].disable();
+                constraints[i] = null;
+            } finally {
+                Constraint.current = null;
+            }
+        }
+        var min = Number.MAX_VALUE;
+        var minIndex = -1;
+        for (var i = 0; i < constraints.length; i++) {
+            if (constraints[i] && constraints[i].oComparisonMetrics.time < min) {
+                min = constraints[i].oComparisonMetrics.time;
```

Comment:
one of the PRs was supposed to get rid of the hungarian notation... I haven't seen the change in the other one